src/main/scala/devices/spi/TLSPI.scala

```diff
@@ -74,36 +74,36 @@ class SPITopModule(c: SPIParamsBase, outer: TLSPIBase)
   io_int(0) := (ip.txwm && ie.txwm) || (ip.rxwm && ie.rxwm)
 
   protected val regmapBase = Seq(
-    SPICRs.sckdiv -> Seq(RegField(c.divisorBits, ctrl.sck.div)),
-    SPICRs.sckmode -> Seq(
-      RegField(1, ctrl.sck.pha),
-      RegField(1, ctrl.sck.pol)),
-    SPICRs.csid -> Seq(RegField(c.csIdBits, ctrl.cs.id)),
-    SPICRs.csdef -> ctrl.cs.dflt.map(x => RegField(1, x)),
-    SPICRs.csmode -> Seq(RegField(SPICSMode.width, ctrl.cs.mode)),
-    SPICRs.dcssck -> Seq(RegField(c.delayBits, ctrl.dla.cssck)),
-    SPICRs.dsckcs -> Seq(RegField(c.delayBits, ctrl.dla.sckcs)),
-    SPICRs.dintercs -> Seq(RegField(c.delayBits, ctrl.dla.intercs)),
-    SPICRs.dinterxfr -> Seq(RegField(c.delayBits, ctrl.dla.interxfr)),
-
-    SPICRs.fmt -> Seq(
-      RegField(SPIProtocol.width, ctrl.fmt.proto),
-      RegField(SPIEndian.width, ctrl.fmt.endian),
-      RegField(SPIDirection.width, ctrl.fmt.iodir)),
-    SPICRs.len -> Seq(RegField(c.lengthBits, ctrl.fmt.len)),
+    SPICRs.sckdiv -> Seq(RegField(c.divisorBits, ctrl.sck.div, RegFieldDesc("sckdiv","Serial clock divisor", reset=Some(3)))),
+    SPICRs.sckmode ->  RegFieldGroup("sckmode",Some("Serial clock mode"),Seq[RegField](
+      RegField(1, ctrl.sck.pha, RegFieldDesc("sckmode_pha","Serial clock phase", reset=Some(0))),
+      RegField(1, ctrl.sck.pol, RegFieldDesc("sckmode_pol","Serial clock polarity", reset=Some(0))))),
+    SPICRs.csid -> Seq(RegField(c.csIdBits, ctrl.cs.id, RegFieldDesc("csid","Chip select id", reset=Some(0)))),
+    SPICRs.csdef -> ctrl.cs.dflt.map(x => RegField(1, x, RegFieldDesc("csdef","Chip select default", reset=Some(1)))),
+    SPICRs.csmode -> Seq(RegField(SPICSMode.width, ctrl.cs.mode, RegFieldDesc("csmode","Chip select mode", reset=Some(SPICSMode.Auto.litValue())))),
+    SPICRs.dcssck -> Seq(RegField(c.delayBits, ctrl.dla.cssck, RegFieldDesc("cssck","CS to SCK delay", reset=Some(1)))),
+    SPICRs.dsckcs -> Seq(RegField(c.delayBits, ctrl.dla.sckcs, RegFieldDesc("sckcs","SCK to CS delay", reset=Some(1)))),
+    SPICRs.dintercs -> Seq(RegField(c.delayBits, ctrl.dla.intercs, RegFieldDesc("intercs","Minimum CS inactive time", reset=Some(1)))),
+    SPICRs.dinterxfr -> Seq(RegField(c.delayBits, ctrl.dla.interxfr, RegFieldDesc("interxfr","Minimum interframe delay", reset=Some(0)))),
+
+    SPICRs.fmt -> RegFieldGroup("fmt",Some("Serial frame format"),Seq[RegField](
+      RegField(SPIProtocol.width, ctrl.fmt.proto, RegFieldDesc("proto","SPI Protocol", reset=Some(SPIProtocol.Single.litValue()))),
+      RegField(SPIEndian.width, ctrl.fmt.endian, RegFieldDesc("endian","SPI Endianness", reset=Some(SPIEndian.MSB.litValue()))),
+      RegField(SPIDirection.width, ctrl.fmt.iodir, RegFieldDesc("iodir","SPI I/O Direction", reset=Some(SPIDirection.Rx.litValue()))))),
+    SPICRs.len -> Seq(RegField(c.lengthBits, ctrl.fmt.len, RegFieldDesc("len","Number of bits per frame", reset=Some(math.min(c.frameBits, 8))))),
 
     SPICRs.txfifo -> NonBlockingEnqueue(fifo.io.tx),
```

Comment:
No way to add descriptions for these? Is `NonBlocking*queue` in this file or some global thing?

Reply:
They are in util/RegMapFIFO.scala - I will add descriptions there and put groups here.